pkg/server/authentication.go

```diff
@@ -147,12 +149,69 @@ func (s *authenticationServer) UserLogin(
 }
 
 // UserLogout allows a user to terminate their currently active session.
+//
+// NOTE: this is an unused stub; we're using the `doLogout` method below instead of
+// this because of https://github.com/grpc-ecosystem/grpc-gateway/issues/470
 func (s *authenticationServer) UserLogout(
 	ctx context.Context, req *serverpb.UserLogoutRequest,
 ) (*serverpb.UserLogoutResponse, error) {
 	return nil, status.Errorf(codes.Unimplemented, "Logout method has not yet been implemented.")
 }
 
+// doLogout allows a user to terminate their currently active session.
+func (s *authenticationServer) doLogout(writer http.ResponseWriter, req *http.Request) {
+	ctx := req.Context()
+
+	sessionID := ctx.Value(webSessionIDKey{})
+	if sessionID == nil {
+		http.Error(writer, "tried to log out without being logged in", http.StatusBadRequest)
+		return
+	}
+
+	// Revoke the session.
+	if n, err := s.server.internalExecutor.Exec(
+		ctx,
+		"revoke-auth-session",
+		nil, /* txn */
+		`UPDATE system.web_sessions SET "revokedAt" = now() WHERE id = $1`,
+		sessionID,
+	); err != nil {
+		http.Error(writer, err.Error(), http.StatusInternalServerError)
+		return
+	} else if n == 0 {
+		http.Error(writer, fmt.Sprintf("session with id %d nonexistent or already revoked", sessionID), http.StatusBadRequest)
+		return
+	}
+
+	// Send back a header which will cause the browser to destroy the cookie.
+	// See https://tools.ietf.org/search/rfc6265, page 7.
+	cookie := makeCookieWithValue("")
+	cookie.MaxAge = -1
+
+	writer.Header().Add("set-cookie", cookie.String())
+
+	acceptContentType := req.Header.Get(httputil.AcceptHeader)
```

Comment:
The rest of this method from here down is just to send a logout response (which is empty!). If we can't use grpc-gateway right now, would at least be good to reuse some kind of response marshaller to avoid writing all of this.